### PR TITLE
Adjusting tcp min read chunk size growth to reduce wastage of incoming buffer size

### PR DESCRIPTION
**Pull request by Vignesh2208**
Double the tcp min chunk size when a recvmsg indicates more data is available to read immediately. This reduces wasted space in the incoming read buffer.

@markdroth

### CODE REVIEW

File: src/core/lib/iomgr/tcp_posix.cc [hunk 1]
```diff
@@ -836,11 +844,21 @@ static void maybe_make_read_slices(grpc_tcp* tcp) {
     int target_length = static_cast<int>(tcp->target_length);
     int extra_wanted =
         target_length - static_cast<int>(tcp->incoming_buffer->length);
+    if (tcp->last_read_completed) {
+      // Set it to false again to start the next block of reads
+      tcp->last_read_completed = false;
+      // Reset last_min_read_chunk_size for the next block of reads
+      tcp->last_min_read_chunk_size = tcp->min_read_chunk_size;
```
Comment (ananda1066) [inline src/core/lib/iomgr/tcp_posix.cc:851]: nit: maybe rename `curr_min_read_chunk_size` and `starting_min_read_chunk_size` or something similar?

Comment (Vignesh2208) [reply src/core/lib/iomgr/tcp_posix.cc]: Changed it now to curr_min_read_chunk_size and curr_read_completed

**Review by ananda1066 — APPROVED** [review]

File: src/core/lib/iomgr/tcp_server_posix.cc [hunk 2]
```diff
@@ -61,6 +61,8 @@
 #include "src/core/lib/iomgr/unix_sockets_posix.h"
 #include "src/core/lib/resource_quota/api.h"
 
+static gpr_atm num_dropped_connections = 0;
```
Comment (ctiller) [inline src/core/lib/iomgr/tcp_server_posix.cc:64]: let's use `std::atomic`

Comment (Vignesh2208) [reply src/core/lib/iomgr/tcp_server_posix.cc]: Done.

**Pull request merged**